File: Arrays_101/Max Consecutive Ones/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function (nums) {
  let maxLength = 0;
  let count = 0;

  nums.forEach((num) => {
    if (num) count++;
    else count = 0;

    maxLength = maxLength < count ? count : maxLength;
  });
  return maxLength;
};

File: Arrays_101/Max Consecutive Ones/solution.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxLength = 0;
        int count = 0;

        for(int num : nums){
            
            if(num) count++;
            else count = 0;

            if(maxLength<count) maxLength = count;
                
        }
            
        return maxLength;
    }
};